openage/launcher/launcher.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2020-2020 the openage authors. See copying.md for legal info.
+
+"""
+Backend logic of the openage launcher
+"""
+
+from .mod import ModInfo
+
+LAUNCHER_VERSION = "0.1.0"
+
+
+class LauncherCore:
+    """
+    Holds the launcher status and provides methods for its functionalities
+    exposes an API for the launcher UI
+    """
+    default_path = 'a/path'
+    default_repo = 'a.repo.com'
+    def __init__(self):
+        self.launcher_version = LAUNCHER_VERSION
+        self.version = self.get_version()
+        if self.is_outdated():
+            print('hey, update!')
+
+        self.local_mods = set()
+        self.remote_mods = set()
+
+        self.import_local_mods(self.default_path)
+        self.import_remote_mods(self.default_repo)
+
+        # store uid of enabled mods, ordered by priority
+        self.enabled = []
+
+    @staticmethod
+    def get_version(self):
+        """
+        returns current openage version
+        """
+        return "0.0.0"
+
+    @staticmethod
+    def get_latest_version(self):
+        """
+        returns the latest available openage version
+        """
+        return "0.0.0"
+
+    def is_outdated(self):
+        if self.version < self.get_latest_version():    # TODO: pseudocode
+            return True
+        return False
+
+    def import_local_mod(self, file_path):
+        mod = ModInfo(file_path=file_path)
+        self.local_mods.add(mod)
+
+    def import_remote_mod(self, url):
+        mod = ModInfo(url=url)
+        self.remote_mods.add(mod)
```

Comment:
Specifying a URL here will give us a lot of security issues. The Launcher should rather load a list of repo URL from a (signed) file, check the signature and then search the existing repos for the UID of a mod.

Reply:
I never had to think about security before :P So, how would you handle this at the ModInfo level? Just feed it something like a file object on init, and handle the rest at a higher level?

Reply:
I don't think ModInfo should handle this. It'd be better to implement a dedicated mod manager that handles adding and updating mods, similar to what Linux distro package managers like apt do ([example](https://wiki.debian.org/SecureApt)). The launcher should just provide a GUI for it. 

If you have never build secure applications before, it's probably better to leave this out for now :D

Reply:
Here is an example from mod.io docs and its' pzthon wrapper:
https://github.com/ClementJ18/mod.io/#example

Reply:
I'm leaving this out for now, assuming the `ModInfo` is initialized with the already extracted toml string, which should be easy to change down the line.